src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+      newPageSelected = this.hasHyperlink
+        ? (elementRef as HTMLAnchorElement)?.href
+        : elementRef?.id;
+    } else {
+      newPageSelected = (elementRef as HTMLButtonElement)?.value;
+    }
+
+    // Don't fire the selectedItemChange event if the button is already selected
+    if (this.selectedPage === newPageSelected) {
+      return;
+    }
+    this.selectedPage = newPageSelected;
+
+    const eventInfo = this.pageChange.emit(this.actualPage);
+
+    if (!eventInfo.defaultPrevented) {
+      this.selectedPage = newPageSelected;
+    }
+  };
+
+  #renderItemsPerPage = () => {
+    return (
+      <form
+        class="form"
+        part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.ITEMS_PER_PAGE]: true })}
+      >
+        <label htmlFor="items-per-page">
+          {this.translations.text.itemsPerPage}
+        </label>
+        <ch-combo-box-render
+          class="combo-box"
+          id="items-per-page"
+          name="items-per-page"
+          model={this.itemsPerPageOptions}
+          onChange={this.#handleItemsPerPageChange}
+          hostParts={tokenMap({
+            [PAGINATOR_PARTS_DICTIONARY.ITEMS_PER_PAGE__COMBO_BOX]: true
+          })}
+          value={this.itemsPerPage.toString()}
+          exportparts={COMBO_BOX_EXPORT_PARTS}
+        />
+      </form>
+    );
+  };
+
+  #renderItemsPerPageInfo = () => {
+    return (
+      <div part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.INFO]: true })}>
+        <span
+          part={tokenMap({ [PAGINATOR_PARTS_DICTIONARY.INFO__TEXT]: true })}
+        >
+          {this.translations.text.showingItems
+            .replace(AMOUNT_ITEMS_IN_PAGE, this.itemsPerPage.toString())
+            .replace(
+              TOTAL_ITEMS,
+              this.totalItems.toString() || this.translations.text.unknownItems
+            )}
+        </span>
+      </div>
+    );
+  };
+
+  #renderFirstControl = () => {
+    return (
+      <button
+        part={tokenMap({
+          [PAGINATOR_PARTS_DICTIONARY.FIRST]: true,
+          [PAGINATOR_PARTS_DICTIONARY.DISABLED]: this.actualPage === 1
+        })}
+        disabled={this.actualPage === 1}
+        onClick={this.#handleFirst}
+      >
+        {this.translations.text.first}
+      </button>
+    );
+  };
+
+  #renderLastControl = () => {
+    return (
+      <button
+        part={tokenMap({
+          [PAGINATOR_PARTS_DICTIONARY.LAST]: true,
+          [PAGINATOR_PARTS_DICTIONARY.DISABLED]:
+            this.actualPage === this.totalPages
+        })}
+        onClick={this.#handleLast}
+        disabled={this.actualPage === this.totalPages}
+      >
+        {this.translations.text.last}
+      </button>
+    );
+  };
+
+  #renderPrevControl = () => {
+    return (
+      <button
+        part={tokenMap({
+          [PAGINATOR_PARTS_DICTIONARY.PREV]: true,
+          [PAGINATOR_PARTS_DICTIONARY.DISABLED]: this.actualPage === 1
+        })}
+        onClick={this.#handlePrev}
```

Comment:
This will also save us memory by not attaching unused handlers.

```suggestion
        onClick={this.actualPage !== 1 && this.#handlePrev}
```